hack/github/get-releases/main.go

```diff
@@ -0,0 +1,192 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"log"
+	"os"
+	"regexp"
+	"slices"
+	"strconv"
+	"strings"
+
+	"github.com/google/go-github/v60/github"
+	"github.com/samber/lo"
+)
+
+const (
+	repoOwner = "aws"
+	repoName  = "karpenter-provider-aws"
+)
+
+type Version struct {
+	Major  int    `json:"major"`
+	Minor  int    `json:"minor"`
+	Patch  int    `json:"patch"`
+	RC     *int   `json:"releaseCandidate,omitempty"`
+	GitTag string `json:"gitTag"`
+	ECRTag string `json:"ecrTag"`
+}
+
+func NewVersionFromTag(tag string) (Version, error) {
+	re, err := regexp.Compile(`v?(\d+)\.(\d+)\.(\d+)(?:-rc\.)?(\d+)?`)
+	if err != nil {
+		return Version{}, err
+	}
+	matches := re.FindAllStringSubmatch(tag, -1)
+	major, err := strconv.Atoi(matches[0][1])
+	if err != nil {
+		return Version{}, err
+	}
+	minor, err := strconv.Atoi(matches[0][2])
+	if err != nil {
+		return Version{}, err
+	}
+	patch, err := strconv.Atoi(matches[0][3])
+	if err != nil {
+		return Version{}, err
+	}
+	var rc *int
+	if matches[0][4] != "" {
+		val, err := strconv.Atoi(matches[0][4])
+		if err != nil {
+			return Version{}, err
+		}
+		rc = &val
+	}
+	result := Version{
+		Major:  major,
+		Minor:  minor,
+		Patch:  patch,
+		RC:     rc,
+		GitTag: tag,
+		ECRTag: tag,
+	}
+	if versionSortFunc(result, Version{
+		Major: 0,
+		Minor: 35,
+		Patch: 0,
+		RC:    nil,
+	}) >= 0 {
+		result.ECRTag = strings.TrimLeft(tag, "v")
+	}
+	return result, nil
+}
+
+var versionSortFunc = func(a, b Version) int {
```

Comment:
Possibly, but not without modification since we need to handle release candidates as well here. I'm not sure it's worth coupling the two tools just to handle version parsing and sorting.